react/fields/MonthPicker/MonthPicker.js

```diff
@@ -0,0 +1,175 @@
+    className: classnames.apply(null, [...classNames, className]), // eslint-disable-line no-useless-call
+    ...restProps
+  };
+}
+
+export default class MonthPicker extends Component {
+
+  static displayName = 'MonthPicker';
+
+  static propTypes = {
+    /* eslint-disable consistent-return */
+    id: (props, propName, componentName) => {
+      const { id, label } = props;
+
+      if (typeof id !== 'string') {
+        return new Error(`Invalid prop \`id\` of type \`${typeof id}\` supplied to \`${componentName}\`, expected \`string\`.`);
+      }
+
+      if (label && !id) {
+        return new Error(`When ${componentName} has a \`label\`, it should also have an \`id\`.`);
+      }
+    },
+    className: PropTypes.string,
+    /* eslint-enable consistent-return */
+    invalid: PropTypes.bool,
+    label: PropTypes.string,
+    /* eslint-disable consistent-return */
+    labelProps: (props, propName, componentName) => {
+      const { id, label, labelProps } = props;
+      const { htmlFor: labelFor } = labelProps || {};
+
+      if (typeof labelProps !== 'undefined' && typeof labelProps !== 'object') {
+        return new Error(`Invalid prop \`labelProps\` of type \`${typeof labelProps}\` supplied to \`${componentName}\`, expected \`object\`.`);
+      }
+
+      if (!label && labelProps) {
+        return new Error(`Specifying \`labelProps\` is redundant when \`label\` is not specified in ${componentName}.`);
+      }
+
+      if (labelFor && id) {
+        return new Error(`\`labelProps.htmlFor\` will be overridden by \`id\` in ${componentName}. Please remove it.`);
+      }
+    },
+    message: PropTypes.string,
+    messageProps: PropTypes.object,
+    value: PropTypes.shape({
+      month: PropTypes.number,
+      year: PropTypes.number
+    }),
+    native: PropTypes.bool,
+    onChange: PropTypes.func,
+    onBlur: PropTypes.func
+  };
+
+  static defaultProps = {
+    id: '',
+    className: '',
+    invalid: false,
+    label: '',
+    message: '',
+    native: false
+  };
+
+  constructor() {
+    super();
+
+    this.renderLabel = this.renderLabel.bind(this);
+    this.renderInput = this.renderInput.bind(this);
+    this.renderMessage = this.renderMessage.bind(this);
+    this.renderIcon = this.renderIcon.bind(this);
+  }
+
+  renderLabel() {
+    const { label } = this.props;
+
+    if (!label) {
+      return;
+    }
+
+    const { labelProps, id } = this.props;
+    const allLabelProps = {
+      ...combineClassNames(labelProps, styles.label),
+      ...(id ? { htmlFor: id } : {})
+    };
+
+    return (
+      <label {...allLabelProps}>
+        {label}
+      </label>
+    );
+  }
+
+  renderInput() {
+    const { id, value, onChange, native, invalid, onBlur } = this.props;
+    const monthPickerProps = {
+      className: styles.input,
+      ...(id ? { id } : {}),
+      value,
+      onChange,
+      onBlur,
+      invalid
+    };
+
+    return (
+      native ?
+        <NativeMonthPicker {...monthPickerProps} /> :
```

Comment:
I got a little confused by the `Native` name, for some reason I thought that meant `ReactNative`. Take from that what you will 😉